src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -101,5 +189,37 @@ public static void Encode(Span<byte> scanline, Span<byte> previousScanline, Span
         /// <returns>The <see cref="int"/></returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static int Average(byte left, byte above) => (left + above) >> 1;
+
+#if SUPPORTS_RUNTIME_INTRINSICS
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<byte> Average(Vector128<byte> left, Vector128<byte> above)
+        {
+            Vector128<ushort> loLeft16 = Sse2.UnpackLow(left, Vector128<byte>.Zero).AsUInt16();
+            Vector128<ushort> hiLeft16 = Sse2.UnpackHigh(left, Vector128<byte>.Zero).AsUInt16();
+
+            Vector128<ushort> loAbove16 = Sse2.UnpackLow(above, Vector128<byte>.Zero).AsUInt16();
+            Vector128<ushort> hiAbove16 = Sse2.UnpackHigh(above, Vector128<byte>.Zero).AsUInt16();
+
+            Vector128<ushort> div1 = Sse2.ShiftRightLogical(Sse2.Add(loLeft16, loAbove16), 1);
+            Vector128<ushort> div2 = Sse2.ShiftRightLogical(Sse2.Add(hiLeft16, hiAbove16), 1);
+
+            return Sse2.PackUnsignedSaturate(div1.AsInt16(), div2.AsInt16());
+        }
```

Comment:
These can be written without widening/narrowing with some bit hackery.  `pavgb` rounds up, but you can make it round down by taking the one's complement of the operands.  i.e. `~pavgb(~left, ~above)` would be the same as `pavgb(left, above)` if it rounded down instead.  I see you're already using `pxor` with an all ones vector in the outer method to perform a vectored bitwise not, so that should slot in easily and run faster.

Reply:
Good catch with `pavgb`. I'll try figuring something out as avoiding all widening and narrowing surely would be an improvement.

Reply:
![image](https://user-images.githubusercontent.com/23627133/118607505-f360b980-b7b8-11eb-9d4e-e93620989d09.png)
That's a nice improvement